### PR TITLE
feat: support hardware keys for playback control

### LINKED ISSUE #2185: Skip track/video previous/ next by Bluetooth
### Describe your suggested feature

Hi,
This app is fantastic, but I found that is not possible to to skip to next/previous track/video when I play a playlist using Bluetooth command (ex when it was used in a car).
Other bug I found:
While I played a playlist, the app stop play after ending a song/video, and not continue 

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by andrew-ld**
you can test this change using `adb shell input keyevent 87` (It should send you to the next video)

closes  #2185

### CODE REVIEW

File: app/src/main/java/com/github/libretube/constants/Constants.kt [hunk 1]
```diff
@@ -47,6 +47,7 @@ const val DOWNLOAD_PROGRESS_NOTIFICATION_ID = 2
 const val DOWNLOAD_CHANNEL_ID = "download_service"
 const val BACKGROUND_CHANNEL_ID = "background_mode"
 const val PUSH_CHANNEL_ID = "notification_worker"
+const val PLAYER_CHANNEL_ID = "player"
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/constants/Constants.kt:50]: The `BACKGROUND_CHANNEL_ID` already serves the purpose of creating the player notification - Therefore you likely want to remove the new `PLAYER_NOTIFICATION_ID` and rename the `BAVKGROUND_NOTIFICATION_ID` to `PLAYER_NOTIFICATION_ID`

**Comment by Bnyro** [comment]
For fixing the lint, please run `ktlint --code_style=android-studio -F` in the root directory of the project.

**Comment by Bnyro** [comment]
Unfortunately, this introduces some major regressions:
* the thumbnail quality in the notification is very low
* it's not possible to seek with the seekbar in the notification
* the notification doesn't show the correct position

It would be great if we can fix all those, otherwise we have to make this behavior optional and let users choose if they prefer the above functionality or support for media buttons.

Otherwise the PR looks great, thanks for your efforts!

**Comment by andrew-ld** [comment]
still need to implement MediaMetadataCompat to report playback information, I will try to do it

**Comment by andrew-ld** [comment]
@Bnyro now seek and location work correctly from the notification

**Comment by andrew-ld** [comment]
for high resolution artwork it seems that NotificationCompat with MediaStyle does not support uri artwork, it only takes the one provided as bitmap which is low quality

**Comment by Bnyro** [comment]
Please rebase with proper git config, you're currently commuting as "ghost", see https://github.com/libre-tube/LibreTube/commit/0557b52ade13a55f751f93ce339019e05073034d.patch

**Comment by andrew-ld** [comment]
for the next configs the commits should appear with my name and github noreply email, now everything should go except the cover image

**Comment by Bnyro** [comment]
Alright, if you share your email here I can rebase the other commits to contain your email as well tomorrow. I assume the git username is the same as on GitHub?

**Comment by andrew-ld** [comment]
@Bnyro I should have solved with git, it seems that duration is not always available from the player, when changing video sometimes I can't get it, I guess I should listen some other event from the player

**Comment by FireMasterK** [comment]
> Alright, if you share your email here I can rebase the other commits to contain your email as well tomorrow. I assume the git username is the same as on GitHub?

You can just use `git log` to find that btw!

**Comment by Bnyro** [comment]
> You can just use `git log` to find that btw!
>
All commits were done as ghost at that point of time :p

File: app/src/main/java/com/github/libretube/util/NowPlayingNotification.kt [hunk 2]
```diff
@@ -166,72 +162,161 @@ class NowPlayingNotification(
...
     }
 
-    private fun getCustomActions() = mutableListOf(
-        // disabled and overwritten in onPlayerCommandRequest
-        // createMediaSessionAction(R.drawable.ic_prev_outlined, PREV),
-        createMediaSessionAction(R.drawable.ic_next_outlined, NEXT),
-        createMediaSessionAction(R.drawable.ic_rewind_md, REWIND),
-        createMediaSessionAction(R.drawable.ic_forward_md, FORWARD)
-    )
+    private fun getMetadataFromPlayer(metadata: MediaMetadata): MediaMetadataCompat {
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/util/NowPlayingNotification.kt:254]: This should be an extension function for MediaMetadata:
```kotlin
fun MediaMetada.toMediaMetadataCompat(): MediaMetadataCompat {
...
}
```
in the `extensions` package.

**Comment by Bnyro** [comment]
I'll see if I can find a way to fix the thumbnail quality, otherwise we should make this optional and let users choose whether they prefer media button controls handling or a good notification thumbnail quality.

**Comment by Bnyro** [comment]
Found a fix for it now, see https://github.com/libre-tube/LibreTube/pull/4405/commits/698f783df3fcf0efe81e91b25aebf7c4909edb5c :)

From my side, there's nothing merge-blocking left apart from the one comment I left above.

Since there are some major changes with this, I'd appreciate if we could get some testers first before merging:

@akano12, @luckkmaxx, @Silther, @rozari0, @IndusAryan: Could you please test the CI / debug builds from this PR for regressions in the player notification when you have some time? Thank you in advance, everyone!

**Comment by akano12** [comment]
On stable and when fast forwarding, the play button changes into circular progress indicator while on this it changes to pause then play again, it was better and more informative previously (specifically on slow networks)


Also there's a bug (not a regression as it also exists on stable and debug). If you play a video and skip to next video grom notification controls, the video will play from the same position of the previous video (not sure if you'd like to work on this issue before merging).



Will edit if i find any issue

**Comment by Bnyro** [comment]
> Also there's a bug (not a regression as it also exists on stable and debug). If you play a video and skip to next video from notification controls, the video will play from the same position of the previous video (not sure if you'd like to work on this issue before merging).
>
That's not related to the changes in this PR, so that should be covered in an other issue/PR.

> On stable and when fast forwarding, the play button changes into circular progress indicator while on this it changes to pause then play again, it was better and more informative previously (specifically on slow networks)
>
In the notification or the normal player?

**Comment by akano12** [comment]
> In the notification or the normal player?

Notification control

**Comment by Bnyro** [comment]
I think I know what's the issue then, will look into it later.

**Comment by andrew-ld** [comment]
the buffering state of exoplayer should be communicated to mediasession, this state on mediasession side is called STATE_BUFFERING (it is related to playback state), I can try to implement it tomorrow, if someone wants to do it today is welcome

**Comment by Bnyro** [comment]
That's what I thought of too, I'll implement that later today.

**Comment by Bnyro** [comment]
Still not 100% fixed, just quickly pushed my changes as I need to leave now

**Comment by Bnyro** [comment]
Please only report things that behave different from the stable release here, the PR doesn't aim to fix other existing bugs.

**Comment by andrew-ld** [comment]
@Bnyro from my side I think there must not be any other change since the bugs reported so far are present on master as well, if you finished the buffering part I think I can remove the draft status to make it to be mergiable

**Comment by Bnyro** [comment]
@andrew-ld yes, agree. The buffering part is still not working properly, and I probably won't continue work on it too soon, so feel free to finish that up. Then it's good to merge from my side.

**Comment by andrew-ld** [comment]
@Bnyro can you tell me in which cases your code regarding buffering does not work

**Comment by andrew-ld** [comment]
I unified the logic to update the playback state, now it should no longer update to an unwanted state

**Comment by Bnyro** [comment]
I didn't know we can override `onIsLoadingChanged`, looks much better from a code perspective now. I'll test the changes later and merge it if I don't find any further issues.

**Comment by Bnyro** [comment]
I fixed one issue I found, seems to work well now.

**Comment by Bnyro** [comment]
Thanks for your work!

**Comment by akano12** [comment]
There's still an issue with play button in the notifications, I'm not sure if this was introduced by this pr. 


To reproduce

1. Play a video
2. Pause the video from the player buttons.
3. Tap home button.

If you open the notifications, the pause/play button would show circular progress indicator as if the video was buffering

**Comment by Bnyro** [comment]
Likely related, please move that to a new issue.

**Pull request merged**